lib/storage/adapters/fs/index.js

```diff
@@ -0,0 +1,123 @@
+var fs = require('fs-extra');
+var path = require('path');
+var url = require('url');
+var Promise = require('es6-promise').Promise;
+var sanitize = require('sanitize-filename');
+
+
+function ensurePath (path) {
+	// Ensure that the specified path exists and is writable. This is quick and
+	// happens on server startup, so sync functions are ok.
+	try {
+		// accessSync throws if the item doesn't exist or we don't have
+		// permission to read + write it.
+		fs.accessSync(path, fs.R_OK | fs.W_OK);
+
+		if (!fs.statSync(path).isDirectory()) {
+			throw Error('Specified output path is not a directory');
+		}
+	} catch (e) {
+		if (e.code === 'ENOENT') {
+			// Recover by creating the directory.
+			fs.mkdirsSync(path);
+			console.log('Storage output path \'' + path + '\' created');
+			return;
+		}
+		throw e;
+	}
+}
+
+function FSAdapter (options, schema) {
+	if (!schema.filename) throw Error('Cannot use FSAdapter without storing filename');
+
+	this.options = options.fs || {};
+	console.log('FSAdapter', this.options);
```

Comment:
Right, so adding some useful debug calls like when files are added and removed would be nice :)